tests/unit/core/test_imageUtils.js

```diff
@@ -0,0 +1,16 @@
+
+import { getAddonIconUrl } from 'core/imageUtils';
+import { fakeAddon } from 'tests/unit/amo/helpers';
+import fallbackIcon from 'amo/img/icons/default-64.png';
+
+describe('getAddonIconUrl', () => {
+  const allowedIcon = 'https://addons.cdn.mozilla.net/webdev-64.png';
+
+  it('return icon url as in fake addon', () => {
+    expect(getAddonIconUrl({ ...fakeAddon, icon_url: allowedIcon })).toEqual(allowedIcon);
+  });
+
+  it('return fallback icon in case of non allowed origin', () => {
+    expect(getAddonIconUrl({ ...fakeAddon, icon_url: 'https://xyz.com/a.png' })).toEqual(fallbackIcon);
+  });
+});
```

Comment:
This needs one more test now that makes sure a falsy add-on (like `getAddonIconUrl(null)`) will return `fallbackIcon`